engine/engine.go

```diff
@@ -0,0 +1,85 @@
+package engine
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/filecoin-project/dealbot/config"
+	"github.com/filecoin-project/dealbot/controller/client"
+	"github.com/filecoin-project/dealbot/tasks"
+
+	logging "github.com/ipfs/go-log/v2"
+)
+
+var log = logging.Logger("engine")
+
+type Engine struct {
+	client *client.Client
+}
+
+func New(cfg *config.EnvConfig) *Engine {
+	workers := 2
```

Comment:
How many concurrent workers we want to have per dealbot to work on tasks - for now I have hardcoded 2, but we can easily make this configurable. I remember a limit of GraphSync of 6, so we might have issues running >6 concurrent retrieval deals - TBD.

Reply:
seems fine